packages/kbn-esql-validation-autocomplete/src/autocomplete/utils/ecs_metadata_helper.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { ESQLRealField } from '../../validation/types';
+
+const removeKeywordSuffix = (name: string) => {
+  return name.endsWith('.keyword') ? name.slice(0, -8) : name;
+};
+
+export interface ECSMetadata {
+  [key: string]: {
+    type?: string;
+    source?: string;
+    description?: string;
+  };
```

Comment:
> [!TIP]
> You could derive this type from [`FieldMetadataPlain`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/fields_metadata/common/index.ts#L13) type from the fields-metadata-plugin, which is directly created from the service that serves these fields to the client.
```ts
export interface ECSMetadata {
  [key: string]: Pick<FieldMetadataPlain, 'description' | 'source' | 'type'>
}
```

Reply:
This unfortunately was a workaround due to circular imports from fields-metadata-plugin into this package.

Reply:
AFAIK the fields metadata doesn't have any deps, how would the cyclic dependency happen here?

Reply:
It's due to imports from multiple package.

```
 proc [tsc] error TS6202: Project references may not form a circular graph. Cycle detected: /Users/quynh/Documents/projects/kibana/packages/kbn-esql-validation-autocomplete/tsconfig.type_check.json
 proc [tsc] /Users/quynh/Documents/projects/kibana/x-pack/plugins/fields_metadata/tsconfig.type_check.json
 proc [tsc] /Users/quynh/Documents/projects/kibana/src/core/tsconfig.type_check.json
 proc [tsc] /Users/quynh/Documents/projects/kibana/packages/core/root/core-root-server-internal/tsconfig.type_check.json
 proc [tsc] /Users/quynh/Documents/projects/kibana/packages/core/apps/core-apps-server-internal/tsconfig.type_check.json
 proc [tsc] /Users/quynh/Documents/projects/kibana/packages/kbn-monaco/tsconfig.type_check.json
```